File: mobile/lib/services/api_service.dart
import 'package:http/http.dart' as http;
import 'dart:convert';

class ApiService {
  final String baseUrl = 'http://localhost:5000/api';

  Future<http.Response> registerUser(Map<String, dynamic> user) async {
    return await http.post(
      Uri.parse('$baseUrl/user/register'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(user),
    );
  }

  Future<http.Response> searchMatches(String userId) async {
    return await http.get(Uri.parse('$baseUrl/matchmaking/search/$userId'));
  }

  Future<http.Response> sendProposal(Map<String, dynamic> proposal) async {
    return await http.post(
      Uri.parse('$baseUrl/matchmaking/propose'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(proposal),
    );
  }
}

File: website_backup/src/app/services/api.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { User, Profile } from '../models/api-models';

@Injectable({
  providedIn: 'root'
})
export class ApiService {
  private baseUrl = 'http://localhost:5000/api';

  constructor(private http: HttpClient) {}

  // User endpoints
  registerUser(user: any): Observable<User> {
    return this.http.post<User>(`${this.baseUrl}/user/register`, user);
  }

  getUser(id: string): Observable<User> {
    return this.http.get<User>(`${this.baseUrl}/user/${id}`);
  }

  // Profile endpoints
  createProfile(profile: any): Observable<Profile> {
    return this.http.post<Profile>(`${this.baseUrl}/profile`, profile);
  }

  getProfile(userId: string): Observable<Profile> {
    return this.http.get<Profile>(`${this.baseUrl}/profile/user/${userId}`);
  }

  updateProfile(profile: Profile): Observable<any> {
    return this.http.put(`${this.baseUrl}/profile`, profile);
  }

  // Matchmaking endpoints
  searchMatches(userId: string): Observable<Profile[]> {
    return this.http.get<Profile[]>(`${this.baseUrl}/matchmaking/search/${userId}`);
  }

  sendProposal(proposal: any): Observable<any> {
    return this.http.post(`${this.baseUrl}/matchmaking/propose`, proposal);
  }
}
